libtiny_tui/src/line_split.rs

```diff
@@ -149,4 +160,121 @@ impl LineDataCache {
         }
         self.line_count = Some(temp_count);
     }
+
+    /// Reverses an iteration of calculate_height() by one.
+    /// Used for removing one character at the end of the buffer.
+    pub fn reverse_by_one(&mut self, buffer: &[char], removed_char: char) {
+        // Subtract the cursor line if there is one
+        let mut temp_count = 1;
+        if let Some(line_count) = self.line_count {
+            temp_count = line_count;
+            // If we made space for the cursor, subtract it.
+            if self.has_cursor && self.current_line_length == self.line_width {
+                temp_count -= 1;
+            }
+        }
+        // Helper to go backwards and find the last whitespace index
+        fn find_last_whitespace_idx(buffer: &[char]) -> Option<i32> {
```

Comment:
Originally, I converted `last_whitespace_idx` to a `Vec<Option<i32>>` then ran `bench.rs` and got really bad results. Changing to this solved that.